app/js/block.js

```diff
@@ -90,3 +119,88 @@ Block.dragStart = (event) => {
   event.preventDefault();
   // Add ability to link blocks from here later.
 };
+
+
+class LoopBlock extends Block {
+
+  constructor(x, y) {
+    super(x, y);
+    this.children = [];
+    this.variable = null;
+  }
+
+  getParamsMeta() {
+    return `Block ${this.uid}: loop loop!`;
+  }
+
+  toJSON() {
+    return JSON.stringify({
+      next: this.next,
+      type: 'loop',
+      loc: this.loc,
+      attributes: {
+        children: this.children,
+      },
+    });
+  }
+
+}
+module.exports.LoopBlock = LoopBlock;
+
+
+class ConditionalBlock extends Block {
+
+  constructor(x, y) {
+    super(x, y);
+    this.variable = null;
+    this.onTrue = null;
+    this.onFalse = null;
+  }
+
+  getParamsMeta() {
+    return `Block ${this.uid}: conditionaaal`;
+  }
+
+  toJSON() {
+    return JSON.stringify({
+      next: this.next,
+      type: 'conditional',
+      loc: this.loc,
+      attributes: {
+        condition: this.variable,
+        children: {
+          true: this.onTrue,
+          false: this.onFalse,
+        },
+      },
+    });
+  }
+
+}
+module.exports.ConditionalBlock = ConditionalBlock;
+
+
+class VariableBlock extends Block {
+
+  constructor(x, y) {
+    super(x, y);
+    this.value = null;
+  }
+
+  getParamsMeta() {
+    return `Block ${this.uid}: variable?!`;
+  }
+
+  toJSON() {
```

Comment:
Hmm, we seem to be repeating this alot. Maybe we should have a generic toJSON in the main block class and add to the object is necessary in each child.